### PR TITLE
fix gem_loader for sequel_pg

### PR DESCRIPTION
**Pull request by will**
Without this, if you use `sequel_pg` in your `Gemfile`, `bundle exec srb init` breaks with an undefined constant `Sequel::Postgres`

### CODE REVIEW

**Comment by will** [comment]
`gem_loader` is completely untested. Figuring out a pattern of how to test it with nothing to go on is too much work for me. If this is unacceptable, please close this request.

File: gems/sorbet/lib/gem_loader.rb [hunk 1]
```diff
@@ -371,6 +371,9 @@ class Sorbet::Private::GemLoader
       my_require 'sequel'
       my_require 'sequel/sql'
     end,
+    'sequel_pg' => proc do
+      my_require 'sequel'
```
Comment (ptarjan) [inline gems/sorbet/lib/gem_loader.rb]: do you not also want to `my_require 'sequel_pg'`

**Comment by ptarjan** [comment]
It is actually super easy to test! You make a test by just making a Gemfile in https://github.com/sorbet/sorbet/tree/master/gems/sorbet/test/snapshot/partial

If you could leave a breadcrumb in the places where you looked for how to test this but decided it was untestable that would help the next person stumbling in here. Thanks!

File: gems/sorbet/lib/gem_loader.rb [hunk 2]
```diff
@@ -371,6 +371,15 @@ class Sorbet::Private::GemLoader
...
     end,
+    'sequel_pg' => proc do
+      # sequel_pg assumes that it was required by the adapter class in sequel
+      # (i.e., it's not mean to be required manually). But also, sequel lazily
+      # loads the adapter class depending on the scheme of the database being
+      # connected to. Since 'srb init' never only requires and never connects,
+      # we need to manually load the adapter class ourself, which will then
+      # transitively load sequel_pg
+      my_require 'sequel/adapters/postgres'
+    end,
```
Comment (jez) [inline gems/sorbet/lib/gem_loader.rb]: tl;dr `sequel_pg` is a weird C-extension that's supposed to be mentioned in a Gemfile but isn't required until the user actually connects to a database.

**Comment by will** [comment]
Thanks for taking care of the tests for me @jez 

@partptarjan the bredcrumbs for me not finding the tests was that I searched the test directory for `gem_loader` and that had zero results, and just `loader` in case it was in a comment or test description and none of those results looked relevant.

**Comment by jez** [comment]
Sorry for the delay on this. I'm actually going to remove the test for now, because requiring `sequel_pg` depends on being able to build `pg`'s native extensions in our test environment:

```
  | Installing pg 1.1.4 with native extensions
  | Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
  |  
  | current directory:
  | /Users/admin/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/pg-1.1.4/ext
  | /Users/admin/.rbenv/versions/2.4.3/bin/ruby -r
  | ./siteconf20190624-35666-1s8iclp.rb extconf.rb
  | checking for pg_config... no
  | No pg_config... trying anyway. If building fails, please try again with
  | --with-pg-config=/path/to/pg_config
  | checking for libpq-fe.h... no
  | Can't find the 'libpq-fe.h header
  | *** extconf.rb failed ***
  | Could not create Makefile due to some reason, probably lack of necessary
  | libraries and/or headers.  Check the mkmf.log file for more details.  You may
  | need configuration options.
  |  
  | Provided configuration options:
  | --with-opt-dir
  | --without-opt-dir
  | --with-opt-include
  | --without-opt-include=${opt-dir}/include
  | --with-opt-lib
  | --without-opt-lib=${opt-dir}/lib
  | --with-make-prog
  | --without-make-prog
  | --srcdir=.
  | --curdir
  | --ruby=/Users/admin/.rbenv/versions/2.4.3/bin/$(RUBY_BASE_NAME)
  | --with-pg
  | --without-pg
  | --enable-windows-cross
  | --disable-windows-cross
  | --with-pg-config
  | --without-pg-config
  | --with-pg_config
  | --without-pg_config
  | --with-pg-dir
  | --without-pg-dir
  | --with-pg-include
  | --without-pg-include=${pg-dir}/include
  | --with-pg-lib
  | --without-pg-lib=${pg-dir}/lib
  |  
  | To see why this extension failed to compile, please check the mkmf.log which can
  | be found here:
  |  
  | /Users/admin/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/extensions/x86_64-darwin-18/2.4.0-static/pg-1.1.4/mkmf.log
  |  
  | extconf failed, exit code 1
  |  
  | Gem files will remain installed in
  | /Users/admin/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/pg-1.1.4 for
  | inspection.
  | Results logged to
  | /Users/admin/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/extensions/x86_64-darwin-18/2.4.0-static/pg-1.1.4/gem_make.out
  |  
  | An error occurred while installing pg (1.1.4), and Bundler cannot continue.
  | Make sure that `gem install pg -v '1.1.4' --source 'https://rubygems.org/'`
  | succeeds before bundling.
  |  
  | In Gemfile:
  | sequel_pg was resolved to 1.12.2, which depends on
  | pg
```

and that's not a yak i'm interested in shaving right now.

**Comment by jez** [comment]
I'll merge this when the tests pass.

**Pull request merged**